### PR TITLE
add one/2 and all/2 functions to Ecto.Multi

### PR DESCRIPTION
**Pull request by lohayon**
Hi,
This PR aims to add query operations to the multi. It is useful to lock an entity inside the transaction.

PS: Thank you for your wonderful work. (ecto and all elixir ecosystem)

### CODE REVIEW

File: lib/ecto/multi.ex [hunk 1]
```diff
@@ -397,6 +397,79 @@ defmodule Ecto.Multi do
     run(multi, name, operation_fun({:delete, fun}, opts))
   end
 
+  @doc """
+  Adds a one operation to the multi.
+  It is usefull to lock an entity inside the transaction.
+  Accepts the same arguments and options as `c:Ecto.Repo.one/2` does.
+  ## Example
```
Comment (josevalim) [inline lib/ecto/multi.ex:404]:
```suggestion
  Runs a query expecting one result in the multi.

  Accepts the same arguments and options as `c:Ecto.Repo.one/2`.

  ## Example

```

File: lib/ecto/multi.ex [hunk 2]
```diff
@@ -397,6 +397,79 @@ defmodule Ecto.Multi do
...
+  def one!(multi, name, queryable, opts) do
+    run(multi, name, operation_fun({:one!, fn _ -> queryable end}, opts))
+  end
+
+  @doc """
+  Adds an all operation to the multi.
+  It is usefull to lock entities inside the transaction.
+  Accepts the same arguments and options as `c:Ecto.Repo.all/2` does.
+  ## Example
+      Ecto.Multi.new()
```
Comment (josevalim) [inline lib/ecto/multi.ex:453]:
```suggestion
  Runs a query in the multi and stores all entries.

  Accepts the same arguments and options as `c:Ecto.Repo.all/2` does.

  ## Example

      Ecto.Multi.new()
```

**Comment by josevalim** [comment]
I pushed some small fixes to the docs and I will merge it once CI passes. Thank you!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by lohayon** [comment]
Hi, I saw `one!` has been removed.
So the function at line https://github.com/elixir-ecto/ecto/blob/v3.8.4/lib/ecto/multi.ex#L860, has no more use.

**Comment by josevalim** [comment]
Great catch, thank you, fixed too!

**Pull request merged**